src/thriveopedia/Thriveopedia.cs

```diff
@@ -618,24 +618,50 @@ private void OnForwardPressed()
 
     private void OnSearchUpdated(string newText)
     {
-        pageTree.Clear();
-        pageTree.CreateItem();
+        stageDropdown.Visible = false;
 
-        // Need to copy to new list as loop modifies the list
-        var pageList = allPages.Keys.ToList();
-        foreach (var page in pageList)
+        foreach (var page in allPages)
         {
-            var children = GetAllChildren(page);
+            var visible = page.Key.TranslatedPageName.ToLower().Contains(newText.ToLower());
```

Comment:
This should specify the culture explicitly. Maybe invariant culture would be preferred or maybe as this is a purely user facing thing this would be a case where current UI culture should be used.

Reply:
Would this be the right way to do this?
`.ToLower(CultureInfo.CurrentCulture)`

I don't think using invariant culture makes sense here as the search should be culture-based